src/transforms/dedupe.rs

```diff
@@ -0,0 +1,269 @@
+use super::Transform;
+use crate::{
+    event::{Event, Value},
+    runtime::TaskExecutor,
+    topology::config::{DataType, TransformConfig, TransformDescription},
+};
+use bytes::Bytes;
+use chrono::{DateTime, Utc};
+use lru::LruCache;
+use serde::{Deserialize, Serialize};
+use std::any::TypeId;
+use string_cache::DefaultAtom as Atom;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct FieldMatchConfig {
+    #[serde(rename = "match")]
+    pub match_fields: Vec<Atom>,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct CacheConfig {
+    pub num_entries: usize,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct DedupeConfig {
+    #[serde(default = "default_field_match_config")]
```

Comment:
Feels like it would have been better to make this a default of `FieldMatchConfig` instead of `DedupeConfig`?

Reply:
So I tried that first, but I guess because it's a nested field doing that would still cause an error if the config file omits the "fields.match" argument, rather than just using the default: 

`ERROR vector: Configuration error: "config/consoleTest.toml": missing field `fields` for key `transforms.deduper``

☹️